### ISSUE TITLE
pipeline.save_pretrained is not saving as safetensors - safe_serialization=True

### ISSUE THREAD

**Issue by FurkanGozukara** [huggingface/diffusers#5763]
I am trying to make a LoRA LCM merger gradio app. 

**I guess there is no way to extract LCM LoRA from pre-trained model? Am I correct?**

It saves everything but as diffusers not as single safetensors file how to fix?

```
    if huggingface:
        print("Loading model from Hugging Face.")
        pipeline = AutoPipelineForText2Image.from_pretrained(model, torch_dtype=torch.float16, variant="fp16")
    else:
        print("Loading model from file.")
        if sdxl or ssd_1b:
            pipeline = StableDiffusionXLPipeline.from_single_file(model)
        else:
            pipeline = StableDiffusionPipeline.from_single_file(model)

    print("Model loading completed.")
    print("Setting up LCMScheduler.")
    pipeline.scheduler = LCMScheduler.from_config(pipeline.scheduler.config)

    print("Loading LoRA weights.")
    if sdxl:
        pipeline.load_lora_weights("latent-consistency/lcm-lora-sdxl")
    elif ssd_1b:
        pipeline.load_lora_weights("latent-consistency/lcm-lora-ssd-1b")
    else:
        pipeline.load_lora_weights("latent-consistency/lcm-lora-sdv1-5")
    print("LoRA weights loaded.")

    print("Fusing LoRA layers.")
    pipeline.fuse_lora(lora_scale=lora_scale)

    if no_half:
        print("Converting pipeline to FP32.")
        pipeline = pipeline.to(dtype=torch.float32)
    else:
        print("Maintaining pipeline as FP16.")
        pipeline = pipeline.to(dtype=torch.float16)

    print(f"Preparing to save the model in {output_path}.")
    os.makedirs(f"{output_path}", exist_ok=True)
    if no_half:
        pipeline.save_pretrained(f"{output_path}/snapshots/{name}", safe_serialization=True)
    else:
        pipeline.save_pretrained(f"{output_path}/snapshots/{name}", variant="fp16", safe_serialization=True)

    print("Model conversion and saving completed.")
    return f"Model {name} converted and saved in {output_path}."
```

the output is as below

![image](https://github.com/huggingface/diffusers/assets/19240467/e7ab8ad1-af35-4c20-b714-bcafd473d960)

![image](https://github.com/huggingface/diffusers/assets/19240467/ab7ff274-78aa-4eda-8daf-6dfb55b1ae2a)

venv below

```
Microsoft Windows [Version 10.0.19045.3636]
(c) Microsoft Corporation. All rights reserved.

G:\gradio_LCM_maker\venv\Scripts>activate

(venv) G:\gradio_LCM_maker\venv\Scripts>pip freeze
accelerate==0.24.1
aiofiles==23.2.1
altair==5.1.2
annotated-types==0.6.0
antlr4-python3-runtime==4.9.3
anyio==3.7.1
attrs==23.1.0
certifi==2022.12.7
charset-normalizer==2.1.1
click==8.1.7
colorama==0.4.6
contourpy==1.2.0
cycler==0.12.1
diffusers==0.23.0
exceptiongroup==1.1.3
fastapi==0.104.1
ffmpy==0.3.1
filelock==3.9.0
fonttools==4.44.0
fsspec==2023.10.0
gradio==4.2.0
gradio_client==0.7.0
h11==0.14.0
httpcore==1.0.2
httpx==0.25.1
huggingface-hub==0.17.3
idna==3.4
importlib-metadata==6.8.0
importlib-resources==6.1.1
Jinja2==3.1.2
jsonschema==4.19.2
jsonschema-specifications==2023.7.1
kiwisolver==1.4.5
markdown-it-py==3.0.0
MarkupSafe==2.1.3
matplotlib==3.8.1
mdurl==0.1.2
mpmath==1.3.0
networkx==3.0
numpy==1.24.1
omegaconf==2.3.0
orjson==3.9.10
packaging==23.2
pandas==2.1.3
Pillow==9.3.0
psutil==5.9.6
pydantic==2.4.2
pydantic_core==2.10.1
pydub==0.25.1
Pygments==2.16.1
pyparsing==3.1.1
python-dateutil==2.8.2
python-multipart==0.0.6
pytz==2023.3.post1
PyYAML==6.0.1
referencing==0.30.2
regex==2023.10.3
requests==2.28.1
rich==13.6.0
rpds-py==0.12.0
safetensors==0.4.0
semantic-version==2.10.0
shellingham==1.5.4
six==1.16.0
sniffio==1.3.0
starlette==0.27.0
sympy==1.12
tokenizers==0.14.1
tomlkit==0.12.0
toolz==0.12.0
torch==2.1.0+cu121
torchaudio==2.1.0+cu121
torchvision==0.16.0+cu121
tqdm==4.66.1
transformers==4.35.0
typer==0.9.0
typing_extensions==4.8.0
tzdata==2023.3
urllib3==1.26.13
uvicorn==0.24.0.post1
websockets==11.0.3
zipp==3.17.0

(venv) G:\gradio_LCM_maker\venv\Scripts>
```

@yiyixuxu @sayakpaul @DN6

**Comment by sayakpaul** [comment]
I don't think we have a conversion script for that. save_pretrained() is always about the diffusers format.

**Comment by FurkanGozukara** [comment]
> I don't think we have a conversion script for that. save_pretrained() is always about the diffusers format.

thank you so much for quick reply. what do you suggest me to solve this issue? any other pipeline to convert that diffusers generated files into safe tensors? help apricated

**Comment by sayakpaul** [comment]
I think you are conflating two concepts into one. 

`safetensors` is a file format. 

When you call save_pretrained() with safe_serialization set to True, the components get saved in the safetensors format. You can verify this by going into the folders you showed in the snapshot. 

But I guess you are looking for a single checkpoint file in the safetensors format that has the UNet, text encoder, etc. We don't have a conversion script for that.

**Comment by FurkanGozukara** [comment]
> But I guess you are looking for a single checkpoint file in the safetensors format that has the UNet, text encoder, etc. We don't have a conversion script for that.

yes accurate I am looking for single checkpoint 

so how do people generate them from diffusers?

**Comment by FurkanGozukara** [comment]
i used https://github.com/huggingface/diffusers/blob/main/scripts/convert_diffusers_to_original_sdxl.py

and it worked but nor automatic1111 neither comfyui is able to use the model :/

**Comment by asomoza** [comment]
I don’t think many people feel the need to generate single-file checkpoints from diffusers. One of its major advantages is the ability to save each module separately.

It’s an interesting idea, although I can’t see the specific use case for this since using the Lora for each model saves a lot more space than storing each full LCM model separately.

Anyway, the easiest way that I can think to do this is to use the Kohya library and pass each separated module (text encoders and Unet) to its conversion function. Here’s a snippet of this:

```python
pipeline = StableDiffusionXLPipeline.from_pretrained(
    "path/to/diffusers_model",
    torch_dtype=torch.float16,
    variant="fp16",
    use_safetensors=True,
).to("cuda")
pipeline.scheduler = LCMScheduler.from_config(pipeline.scheduler.config)
pipeline.load_lora_weights("path/to/lcm-lora-sdxl.safetensors")
pipeline.fuse_lora(lora_scale=1.0)
pipeline.unload_lora_weights()

text_encoder1 = pipeline.text_encoder
text_encoder2 = pipeline.text_encoder_2
vae = pipeline.vae
unet = pipeline.unet

del pipeline

state_dict = convert_diffusers_unet_state_dict_to_sdxl(unet.state_dict())
with init_empty_weights():
    unet = sdxl_original_unet.SdxlUNet2DConditionModel()  # overwrite unet

load_state_dict_on_device(unet, state_dict, device="cuda", dtype=torch.float16)

save_stable_diffusion_checkpoint(
    "path/to/model.safetensors",
    text_encoder1,
    text_encoder2,
    unet,
    None,
    None,
    None,
    vae,
    None,
    None,
    torch.float16,
)
```
I did a quick test that worked in ComfyUI, here's the model I converted: https://huggingface.co/OzzyGT/lcm-juggernaut-xl

Just in case, that function is in: https://github.com/bmaltais/kohya_ss/blob/9eb5da23002c6e00f084a19d62da6fc82074c773/library/sdxl_model_util.py#L468
so you'll need to import it from that file.

**Comment by FurkanGozukara** [comment]
I combined with LCM LoRA and generated safetensors but both Automatic1111 and ComfyUI fails to use

I also tested LCM lora on ComfyUI and definitely huge quality loss
